### PR TITLE
M3-xxx Test Config Changes

### PR DESCRIPTION
**Pull request by martinmckenna**
### Description

Change test configs

This fixes two cases

1. Where the e2e tests would attempt to login and get a CSRF Error on the login screen, causing all subsequent tests to bomb because the test runner wouldn't attempt to relogin
2. E2E tests would fail because it was stuck on the Authorize 3rd party app page

### To Test

Run `yarn e2e` and wait until you get the CSRF error. The test runner should attempt another login

### CODE REVIEW

File: e2e/utils/config-utils.js [hunk 1]
```diff
@@ -42,46 +42,79 @@ exports.readToken = (username) => {
...
     browser.waitForVisible('#password', constants.wait.long);
     browser.trySetValue('#username', username);
     browser.trySetValue('#password', password);
-    loginButton = browser.getUrl().includes('dev') ? '.btn#submit' : '.btn-primary';
-    letsGoButton = browser.getUrl().includes('dev') ? '.btn#submit' : '[data-qa-welcome-button]';
+
+    const loginButton = browser.getUrl().includes('dev') ? '.btn#submit' : '[data-qa-sign-in] input';
+    const letsGoButton = browser.getUrl().includes('dev') ? '.btn#submit' : '[data-qa-welcome-button]';
+
+    // Helper to check if on the Authorize 3rd Party App Income
```
Comment (Jskobos) [inline e2e/utils/config-utils.js]: wut

File: e2e/utils/config-utils.js [hunk 2]
```diff
@@ -42,46 +42,79 @@ exports.readToken = (username) => {
...
+            return true;
+        } catch (err) {
+            console.log('Not on the Oauth Page, continuing');
+            return false;
+        }
+    }
+
+    // Helper to check if CSRF error is displayed on the page
+    const csrfErrorExists = () => {
+        const sourceIncludesCSRF = browser.getSource().includes('CSRF');
```
Comment (rulikrulit) [inline e2e/utils/config-utils.js]: Can be simplified with `return browser.getSource().includes('CSRF')` or even `const csrfErrorExists = browser.getSource().includes('CSRF')` but that will change the type of the variable

File: e2e/utils/config-utils.js [hunk 3]
```diff
@@ -42,46 +42,79 @@ exports.readToken = (username) => {
...
-        browser.waitUntil(function() {
-            return browser.isExisting('[data-qa-add-new-menu-button]') || browser.getUrl().includes('oauth/authorize');
-        }, constants.wait.normal);
-    } catch (err) {
-        console.log('failed to login!');
-        if (browser.getText('.alert').includes('This field is required.')) {
+    const onOauthPage = isOauthAuthPage();
+    const csrfError = csrfErrorExists();
+
+    if (onOauthPage || csrfError) {
```
Comment (rulikrulit) [inline e2e/utils/config-utils.js]: This if wrapper is redundant. There is no common code for both cases and both cases are handled separately below

Comment (martinmckenna) [reply e2e/utils/config-utils.js]: yep caught that one

File: e2e/utils/config-utils.js (hunk 1 above)
Comment (patthiel) [reply e2e/utils/config-utils.js]: LOL something in my brain misfired here..

**Pull request merged**